[docs] Improve recipe with `useLayoutEffect` and `beforeunload` event-listener (#10811)
mui/mui-x: docs/data/data-grid/state/state.md

### Save and restore the state from external storage

You can use `apiRef.current.exportState()` to save a snapshot of the state to an external storage (e.g. using `localStorage` or `redux`).
This way the state can be persisted on refresh or navigating to another page. This is done by listening on the `beforeunload` event.

> When used in an SPA the `useLayoutEffect` cleanup function is being used instead.

Reviewer: ```suggestion
When the component is unmounted, the `useLayoutEffect` cleanup function is being used instead.
```

Unmount would be more precise here, as it's not really about SPA 🙂
